src/pybind/mgr/snap_schedule/fs/schedule.py

```diff
@@ -0,0 +1,311 @@
+"""
+Copyright (C) 2020 SUSE
+
+LGPL2.1.  See file COPYING.
+"""
+from datetime import datetime, timezone
+import logging
+import sqlite3
+
+log = logging.getLogger(__name__)
+
+
+class Schedule(object):
+    '''
+    Wrapper to work with schedules stored in sqlite
+    '''
+    def __init__(self,
+                 path,
+                 schedule,
+                 retention_policy,
+                 fs_name,
+                 rel_path,
+                 start=None,
+                 subvol=None,
+                 created=None,
+                 first=None,
+                 last=None,
+                 last_pruned=None,
+                 created_count=0,
+                 pruned_count=0,
+                 active=True,
+                 ):
+        self.fs = fs_name
+        self.subvol = subvol
+        self.path = path
+        self.rel_path = rel_path
+        self.schedule = schedule
+        self.retention = retention_policy
+        if start is None:
+            now = datetime.now(timezone.utc)
+            self.start = datetime(now.year,
+                                  now.month,
+                                  now.day,
+                                  tzinfo=now.tzinfo)
+        else:
+            self.start = datetime.fromisoformat(start).astimezone(timezone.utc)
+        if created is None:
+            self.created = datetime.now(timezone.utc)
+        else:
+            self.created = datetime.fromisoformat(created)
+        if first:
+            self.first = datetime.fromisoformat(first)
+        else:
+            self.first = first
+        if last:
+            self.last = datetime.fromisoformat(last)
+        else:
+            self.last = last
+        if last_pruned:
+            self.last_pruned = datetime.fromisoformat(last_pruned)
+        else:
+            self.last_pruned = last_pruned
+        self.created_count = created_count
+        self.pruned_count = pruned_count
+        self.active = active
+
+    @classmethod
+    def _from_get_query(cls, table_row, fs):
+        return cls(table_row['path'],
+                   table_row['schedule'],
+                   table_row['retention'],
+                   fs,
+                   table_row['rel_path'],
+                   table_row['start'],
+                   table_row['subvol'],
+                   table_row['created'],
+                   table_row['first'],
+                   table_row['last'],
+                   table_row['last_pruned'],
+                   table_row['created_count'],
+                   table_row['pruned_count'],
+                   table_row['active'])
+
+    def __str__(self):
+        return f'''{self.path} {self.schedule} {self.retention}'''
+
+    CREATE_TABLES = '''CREATE TABLE schedules(
+        id integer PRIMARY KEY ASC,
```

Comment:
I know it doesn't matter but you use `INT`, `integer`, and `int` interchangeably. I'd prefer just using `INT`. Same for the other types below...